addOns/pscanrulesAlpha/src/test/java/org/zaproxy/zap/extension/pscanrulesAlpha/InsecureComponentScannerUnitTest.java

```diff
@@ -0,0 +1,408 @@
+        generatorVulnerabilitites.add(new CVE("man in the middle", Double.valueOf(4.0D)));
+        cache.add(
+                new Product(
+                        Product.ProductType.PRODUCTTYPE_CONTENT_GENERATOR,
+                        "",
+                        "Frontweaver",
+                        "1.2.0"),
+                generatorVulnerabilitites);
+
+        LinkedList<CVE> oracleVulnerabilitites = new LinkedList<CVE>();
+        oracleVulnerabilitites.add(new CVE("man in the middle", Double.valueOf(4.0D)));
+        cache.add(
+                new Product(Product.ProductType.PRODUCTTYPE_WEBSERVER, "", "Oracle", "2.4.0"),
+                oracleVulnerabilitites);
+
+        LinkedList<CVE> jettyVulnerabilitites = new LinkedList<CVE>();
+        jettyVulnerabilitites.add(new CVE("man in the middle", Double.valueOf(8.0D)));
+        cache.add(
+                new Product(Product.ProductType.PRODUCTTYPE_WEBSERVER, "", "Jetty", "2.4.0"),
+                jettyVulnerabilitites);
+
+        LinkedList<CVE> jbossVulnerabilitites = new LinkedList<CVE>();
+        jbossVulnerabilitites.add(new CVE("man in the middle", Double.valueOf(4.0D)));
+        cache.add(
+                new Product(Product.ProductType.PRODUCTTYPE_WEBSERVER, "", "JBoss", "5.0.0"),
+                jbossVulnerabilitites);
+
+        LinkedList<CVE> tomcatVulnerabilitites = new LinkedList<CVE>();
+        tomcatVulnerabilitites.add(new CVE("man in the middle", Double.valueOf(4.0D)));
+        cache.add(
+                new Product(Product.ProductType.PRODUCTTYPE_WEBSERVER, "", "Tomcat", "9.1.0"),
+                tomcatVulnerabilitites);
+
+        LinkedList<CVE> proxyVulnerabilitites = new LinkedList<CVE>();
+        proxyVulnerabilitites.add(new CVE("ddos attack", Double.valueOf(4.0D)));
+        cache.add(
+                new Product(Product.ProductType.PRODUCTTYPE_PROXY_SERVER, "", "proxy", "3.0"),
+                tomcatVulnerabilitites);
+    }
+
+    private HttpMessage createMessage() throws URIException, HttpMalformedHeaderException {
+        HttpRequestHeader requestHeader = new HttpRequestHeader();
+        requestHeader.setMethod("GET");
+        requestHeader.setURI(new URI("https://example.com/fred/", false));
+
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader(requestHeader);
+        return msg;
+    }
+
+    @Override
+    protected InsecureComponentScanner createScanner() {
+        return new InsecureComponentScanner();
+    }
+
+    private void assertVulnerabilityAlertRaised(int expectedRiskLevel, String expectedEvidence) {
+        assertThat(alertsRaised.size(), equalTo(1));
+        assertThat(alertsRaised.get(0).getRisk(), equalTo(expectedRiskLevel));
+        assertThat(alertsRaised.get(0).getEvidence(), equalTo(expectedEvidence));
+    }
+
+    @Test
+    public void shouldRaiseVulnerabilityAlertWhenInsecurePHPVersionUsed()
+            throws URIException, HttpMalformedHeaderException {
```

Comment:
Are you sure about URIException or is that a leftover from earlier versions? The tests don’t seem to actually do any URI manipulation. (This goes for other methods as well)

Reply:
`URIException` can be thrown in class `InsecureComponentScannerUnitTest` by `requestHeader.setURI(new URI("https://example.com/fred/", false));` in line 111.
I added the URI to the request header, to prevent `NullPointerException` when setting the alert details in class `InsecureComponentScanner` in line 439:
`msg.getRequestHeader().getURI().toString()`.

Reply:
Not a major issue, just s need out of place. I don’t recall how other unit tests handle this.